src/lib/i18n.js

```diff
@@ -19,6 +19,8 @@ export function useLanguage() {
 		[url]
 	);
 
+	setLang(lang);
```

Comment:
I believe this will cause an infinite promise loop, or at least a lot of duplicate renders.